source/extensions/filters/http/ext_authz/ext_authz.cc

```diff
@@ -199,6 +203,17 @@ void Filter::onComplete(Filters::Common::ExtAuthz::ResponsePtr&& response) {
       }
     }
 
+    ENVOY_STREAM_LOG(trace, "ext_authz filter removed header(s) from the request:", *callbacks_);
+    for (const auto& header : response->headers_to_remove) {
+      // We don't allow removing any :-prefixed headers, nor Host, as removing
+      // them would make the request malformed.
+      if (header.get()[0] == ':' || header == Http::Headers::get().HostLegacy) {
```

Comment:
This works, but let's use `absl::StartsWithIgnoreCase` instead?

Reply:
Fixed